src/components/Project/ProjectBasics/ProjectForm/ProjectForm.tsx

```diff
@@ -54,6 +56,116 @@ const ProjectForm = () => {
+
+    financesCopy = { ...financesCopy, [startYearName]: newBudget.toFixed(2)}
+    return financesCopy;
+  }
+
+  const moveBudgetBackwards = (finances: IProjectFinances, previousEndYear: number, endYear: number) => {
+    let financesCopy = finances;
+    const numberOfYears = previousEndYear - endYear;
+    const maxIndex = 10 - (endYear - finances.year);
+    let budgetToMove = 0.00;
+    for (let i=1; i<=numberOfYears && i<=maxIndex; ++i) {
+      const financeYearName = getFinanceYearName(finances, endYear + i);
+      const financeValue = finances[financeYearName];
+      budgetToMove += parseFloat(financeValue as string);
+      financesCopy = { ...financesCopy, [financeYearName]: "0.00"};
+    }
+    const endYearName = getFinanceYearName(finances, endYear);
+    const newBudget = (parseFloat(finances[endYearName] as string) || 0.00) + budgetToMove;
+
+    financesCopy =  { ...financesCopy, [endYearName]: newBudget.toFixed(2)}
+    return financesCopy;
+  }
+
+  const updateFinances = (data: IProjectRequest, project: IProject) => {
+    let finances = project.finances;
+    if (data.planningStartYear) {
+      const planningStartYear = project.planningStartYear;
+      // If new planning start year is bigger than the previous one, budget from years that are not within the schedule of the project
+      // need to be moved to the new planning start year
+      if(planningStartYear && data.planningStartYear > planningStartYear) {
+        const updatedFinances = moveBudgetForwards(finances, planningStartYear, data.planningStartYear);
+        data = {...data, "finances": updatedFinances}
+        finances = updatedFinances;
+      }
+    }
+    if (data.estPlanningEnd) {
+      const previousPlanningEndYear = getYear(project.estPlanningEnd);
+      const planningEndYear = getYear(data.estPlanningEnd);
+      const constructionStartYear = getYear(project.estConstructionStart);
+      // If new planning end year is smaller that the previous one, budget from the years that are not within the schedule need to
+      // be moved backwards to the new end year
+      if (previousPlanningEndYear && planningEndYear && planningEndYear < previousPlanningEndYear) {
+        // If there was an overlap between planning end year and construction start year, budget shouldn't be moved from that year
+        // but still needs to be moved from all the years that are not within the schedule anymore
+        if (constructionStartYear == previousPlanningEndYear &&  previousPlanningEndYear - planningEndYear > 1) {
+          const updatedFinances = moveBudgetBackwards(finances, previousPlanningEndYear - 1, planningEndYear);
+          finances = updatedFinances;
+        } else if (constructionStartYear != previousPlanningEndYear) {
+          const updatedFinances = moveBudgetBackwards(finances, previousPlanningEndYear, planningEndYear);
+          data = {...data, "finances": updatedFinances};
+          finances = updatedFinances;
+        }
```

Comment:
This could be refactored as the only difference in if/else sentences is that -1 with previousPlanningEndYear. I made some refactoring suggestion with a similar part in the code